### PR TITLE
feat(input): display type icons

### PR DESCRIPTION
**Pull request by AykutSarac**
I decided that'd be a good idea to display icons depending on the type of the input.

> `icon` attribute overrides the default icon.

image

```html
<bl-input type="search" />
<bl-input type="date" />
<bl-input type="date" icon="academy" />
<bl-input type="time" />
```

Additionally, this PR adds input type of `search`, it could be good idea to review #712 first.

### CODE REVIEW

**Comment by muratcorlu** [comment]
I think it will be nice to merge #712 first, then add default icons with this PR

**Comment by AykutSarac** [comment]
> I think it will be nice to merge #712 first, then add default icons with this PR

Actually #712 has no crucial relevance with this PR so this could be reviewed separately.

**Comment by ogunb** [comment]
Should we also update the clear button for type search with this PR? Or should there be another issue for it?

**Comment by muratcorlu** [comment]
> Should we also update the clear button for type search with this PR? Or should there be another issue for it?

I think "clear button for input" is another story. Since some inputs (like search) already puts a native clear button in some browsers, but some others don't. Probably we'll need to make a generic decision that covers all types of inputs. It would be better to consider it separately.

**Review by muratcorlu — APPROVED** [review]
It looks good to me.

File: src/components/input/bl-input.ts [hunk 1]
```diff
@@ -284,6 +294,7 @@ export default class BlInput extends FormControlMixin(LitElement) {
 
   firstUpdated() {
     this.setValue(this.value);
+    if (!this.icon) this.icon = inputTypeIcons[this.type];
```
Comment (muratcorlu) [inline src/components/input/bl-input.ts:297]: I think it's better to have a test for this.

Also some examples in storybook would be helpful to check them visually.

**Review by muratcorlu — CHANGES_REQUESTED** [review]
Unit and UI Test would be great.

**Review by buseselvi — APPROVED** [review]

**Review by muratcorlu — APPROVED** [review]
Good job! 🚀

**Review by fatihhayri — APPROVED** [review]

**Pull request merged**